permissions/src/main/java/com/google/accompanist/permissions/MultiplePermissionsState.kt

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.permissions
+
+import androidx.activity.compose.LocalActivityResultRegistryOwner
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.ActivityResultRegistry
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+
+/**
+ * Creates a [MultiplePermissionsState] that is remembered across compositions.
+ *
+ * It's recommended that apps exercise the permissions workflow as described in the
+ * [documentation](https://developer.android.com/training/permissions/requesting#workflow_for_requesting_permissions).
+ *
+ * @param permissions the permissions to control and observe.
+ */
+@Composable
+fun rememberMultiplePermissionsState(
+    vararg permissions: String,
+): MultiplePermissionsState {
+    val activityResultRegistry = LocalActivityResultRegistryOwner.current?.activityResultRegistry
+        ?: throw IllegalStateException()
+    return rememberMultiplePermissionsState(activityResultRegistry, *permissions)
+}
+
+/**
+ * Creates a [MultiplePermissionsState] that is remembered across compositions.
+ *
+ * It's recommended that apps exercise the permissions workflow as described in the
+ * [documentation](https://developer.android.com/training/permissions/requesting#workflow_for_requesting_permissions).
+ *
+ * @param activityResultRegistry to store the permission result callbacks.
+ * @param permissions the permissions to control and observe.
+ */
+@Composable
+fun rememberMultiplePermissionsState(
```

Comment:
Generally I think there's a bit too much happening (and too much being remembered in this  function). Can we push more of it down into the `MultiplePermissionsState` class? That helps testing as you no longer need to run it as a composable.

Reply:
Similar for `PermissionsState` tbh

Reply:
Fixed now. Thanks!